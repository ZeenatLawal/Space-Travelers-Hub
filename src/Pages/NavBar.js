import Header from './Header';

const NavBar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/Missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/My Profile',
      text: 'My Profile',
    },
  ];

  return (
    <nav className="navBar">
      <Header />
      <ul className="navLinks">
        {links.map((link) => (
          <li key={link.id} className="navLink">
            {link.text}
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default NavBar;
