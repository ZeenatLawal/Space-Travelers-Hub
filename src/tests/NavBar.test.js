import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
// eslint-disable-next-line import/extensions
import NavBar from '../pages/NavBar.js';

describe('test if NavBar', () => {
  test('renders correctly', () => {
    const component = renderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('NavBar', () => {
  test('Responding on Rockets click ', () => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('Rockets'));
    screen.debug();
    expect(screen.getByTestId('1'));
  });

  test('Responding on Missions click ', () => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('Missions'));
    screen.debug();
    expect(screen.getByTestId('2'));
  });

  test('Responding on My Profile click ', () => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('My Profile'));
    screen.debug();
    expect(screen.getByTestId('3'));
  });
});
