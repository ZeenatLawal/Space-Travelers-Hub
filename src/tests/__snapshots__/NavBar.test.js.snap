// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if NavBar renders correctly 1`] = `
<nav
  className="navBar"
>
  <div
    className="navWrap"
  >
    <div
      className="headerWrap"
    >
      <img
        alt="haha"
        className="logoImg"
        src="../planet.png"
      />
      <h1>
        Space Traveler's Hub
      </h1>
    </div>
    <ul
      className="navLinks"
    >
      <li
        className="navLink"
      >
        <a
          aria-current="page"
          className="active"
          data-testid={1}
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Rockets
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          aria-current={null}
          data-testid={2}
          href="/Missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          aria-current={null}
          data-testid={3}
          href="/MyProfile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
